docs/testing.md

```diff
@@ -0,0 +1,54 @@
+# Testing Guidelines & Framework
+
+## Testing Framework
+
+[Go](https://golang.org) has extensive support for automated [testing](https://golang.org/pkg/testing/) built into the language.
+
+
+Additionally, we utilize the [Gofight](https://github.com/appleboy/gofight) framework to test API handler mock responses and [Baloo](gopkg.in/h2non/baloo.v3) for end-to-end tests.
+
+## Go Unit Tests
+The unit tests have no external dependencies and can be run like so:
+```shell
+$ go test -v ./... -short
+```
+
+## Go Package Tests
+
+_Note: The db package test is dependent on [Localstack](docs/localstack.md) running as they are testing individual connectivity to
+external services._
+
+```shell
+AWS_REGION=localstack go test -v ./[PACKAGE NAME]/
```

Comment:
This makes sense, however I'm a little confused as to how to know programmatically what packages I need to run. I suppose one option is to just change our testing pattern for every package that we add but that feels a little fragile.

Reply:
How about:
```
go list -f '{{if len .TestGoFiles}}"go test {{.ImportPath}}"{{end}}' ./...|xargs -L 1 sh -c
```

Reply:
Is that really want we want? I have more questions about why something like package tests (unit tests) need an external service running? Can we mock ext services for those?